### ISSUE TITLE
Mix Effect companion connection issue

### ISSUE THREAD

**Issue by AVProductions** [bitfocus/companion-module-tow-mixeffect#25]
Suddenly incapable of connecting to MixEffect from companion: 

Companion [3.1.2 (3.1.2+6243-stable-05c37a10)]
MixEffect: 1.9.1
Mac OS: 14.1.2
Hardware: M3 Max MBP and also M1 Max MPB same issue for both: I was able to connect on December 3 but on December 4 2023 connection is always getting error message: **Connection Failure
TypeError: fetch failed**

I have only one instance of Companion and MixEffect opened at the same time, stream deck plugin disabled, and IP address and port configured correctly as it was working 1 day before. 

Thanks for support!

**Comment by adamtow** [comment]
This is probably because you have the HTTP Feedback Server enabled in Companion. If you're not using it, you can turn it off in the Connection tab. Otherwise, make sure that the Port is matches the Port in MixEffect's Switcher Connection entry for the HTTP Feedback Server.

**Comment by waglerit** [comment]
I just added Mix Effect to Companion and encountered the same problem. It connects to the device, but gives me a subsequent error 500ms later. When I turn off feedback polling, there is no error. I can control Mix Effect just fine through Companion, but there's obviously no feedback.

**Comment by adamtow** [comment]
> I just added Mix Effect to Companion and encountered the same problem. It connects to the device, but gives me a subsequent error 500ms later. When I turn off feedback polling, there is no error. I can control Mix Effect just fine through Companion, but there's obviously no feedback.

Can you confirm that you've entered the right port number in MixEffect's Switcher Connection entry? If the HTTP Server is running on the MixEffect device, it should be able to send/receive feedback from Companion.

**Comment by waglerit** [comment]
Yes. Sorry if I was a bit unclear. I can send commands to Mix Effect through Companion, so I have to assume IP and port are correct (plus I double checked). It's just the feedback polling that's giving me the error. 

I would have changed ports etc., but I cannot do that in Mix Effect, so I stuck to the IP and port that were given. It works just fine for controlling Mix Effect, there is just no feedback.

**Comment by adamtow** [comment]
> Yes. Sorry if I was a bit unclear. I can send commands to Mix Effect through Companion, so I have to assume IP and port are correct (plus I double checked). It's just the feedback polling that's giving me the error.
> 
> I would have changed ports etc., but I cannot do that in Mix Effect, so I stuck to the IP and port that were given. It works just fine for controlling Mix Effect, there is just no feedback.

The IP address would be the same, but you should be able to change the port in the Switcher Connection section. Tap Edit in the upper-right corner to edit the entry, change the port, and tap Save. You may need to stop and restart the connection for changes to take effect. Also make sure that there isn't another device that's using the same feedback port. You can also see if setting a higher polling interval works, like 750ms. What kind of device are you running MixEffect on? Mac, iPad, or iPhone?

**Comment by waglerit** [comment]
Okay. I was able to change the port, to no effect. I increased the polling interval to different values, to no effect. Still getting 

I am using an iPad for MixEffect. Companion is running on a Windows PC. 

![image](https://github.com/bitfocus/companion-module-tow-mixeffect/assets/11270000/9b3d9a30-e62e-4758-9ab8-7bad4727c94b)

**Comment by adamtow** [comment]
> Okay. I was able to change the port, to no effect. I increased the polling interval to different values, to no effect. Still getting
> 
> I am using an iPad for MixEffect. Companion is running on a Windows PC.
> 

If you go to IP_ADDRESS_OF_MIXEFFECT:PORTNUMBER/feedback what happens? For instance:

http://10.0.1.19:8080/feedback where 10.0.1.19 is the IP of MixEffect and 8080 is the port number. If a JSON blob appears, things are working properly. If you get any other error, let me know.

**Comment by PostArchitekt** [comment]
I'm currently having a similar situation. OSC and http server are running. IP addresses are correct in Companion. Elgato software is enabled. MixEffect on iPhone is working. I see my mixeffect streamdeck profile on the streamdeck. Yet, nothing happens when I hit the keys. Any thoughts?

**Comment by adamtow** [comment]
> I'm currently having a similar situation. OSC and http server are running. IP addresses are correct in Companion. Elgato software is enabled. MixEffect on iPhone is working. I see my mixeffect streamdeck profile on the streamdeck. Yet, nothing happens when I hit the keys. Any thoughts?

Does this article help? https://labs.mixeffect.app/article/2022/05/companion-profile-installation-troubleshooting/

**Comment by McMark1441** [comment]
I am encountering this issue as well. Periodically companion will loose connection to mix effects, and sometimes will regain connection upon focusing the mix effects application.

[module-log-1_6_2026-12_58_15 PM.csv](https://github.com/user-attachments/files/24459033/module-log-1_6_2026-12_58_15.PM.csv)

I am actively using feedbacks, so I can't just disable the pulling requests. I've even tried changed the pulling rate to 750ms with no change.

**Comment by l4flirop** [comment]
I have mix effect running on my iPad connected to the atem production studio 4k and its operating fine however when I load companion on my MacBook and attempt to use stream deck after loading the connection I am unable to reach mixeffect.

I am able to reach atem switcher from companion as I have green check mark connection is good but in version 2.0.6 I am unable to reach mix effects any help would be appreciated.

**Comment by adamtow** [comment]
> I have mix effect running on my iPad connected to the atem production studio 4k and its operating fine however when I load companion on my MacBook and attempt to use stream deck after loading the connection I am unable to reach mixeffect.
> 
> I am able to reach atem switcher from companion as I have green check mark connection is good but in version 2.0.6 I am unable to reach mix effects any help would be appreciated.

Can you provide some screenshots of your MixEffect switcher connection entry (showing the port) and your Companion config? If you change something in the switcher connection entry, you sometimes have to disconnect and reconnect for the OSC server to become active. You'll also want to make sure that you don't have multiple MixEffect windows running connected to the same switcher (they will contend for the same OSC port). 

Also make sure that if you're using HTTP feedback to have the right port set in both MixEffect and Companion. Otherwise, uncheck feedback in your Companion config.

**Comment by l4flirop** [comment]
<img width="714" height="370" alt="Image" src="https://github.com/user-attachments/assets/c8e708d9-8aff-44c7-9adf-9f9f478dd71c" />

<img width="742" height="894" alt="Image" src="https://github.com/user-attachments/assets/7d0d79f6-2b82-4f38-9a25-8bb74d9df482" />

**Comment by adamtow** [comment]
Your OSC port is 0. You should set it to what you have set in the MixEffect Switcher Connection entry. I would choose 49990. Make sure you disconnect and reconnect in MixEffect for the server to restart properly.

**Comment by l4flirop** [comment]
I did that this is what's on the iPad  settings 

<img width="2732" height="2048" alt="Image" src="https://github.com/user-attachments/assets/5480f3c0-6302-4566-b69f-c4e6511ea4e7" />

**Comment by l4flirop** [comment]
<img width="423" height="464" alt="Image" src="https://github.com/user-attachments/assets/8894d83f-e082-4aea-8966-d3ede6f0a033" />

**Comment by l4flirop** [comment]
<img width="1547" height="817" alt="Image" src="https://github.com/user-attachments/assets/3121a81a-0a01-4ca1-ab12-d988262ea30d" />

**Comment by adamtow** [comment]
No, don't change the Companion port, change that back to what it previously was (8000?). In the MixEffect module configuration, you change the OSC Port to 49990 and the HTTP Server Port to what's set in MixEffect (the default is 8080).

**Comment by adamtow** [comment]
See screenshots:

<img width="676" height="639" alt="Image" src="https://github.com/user-attachments/assets/e34309e7-e2ba-433f-b690-1c8e9b940903" />

<img width="1290" height="2796" alt="Image" src="https://github.com/user-attachments/assets/77b38bb7-4949-459a-be84-347de5e98bd6" />

**Comment by l4flirop** [comment]
HTTP Server is greyed out on the iPad 

<img width="2732" height="2048" alt="Image" src="https://github.com/user-attachments/assets/aef60e2f-eb13-4174-8913-09ecd57303a9" />

**Comment by adamtow** [comment]
You need to enable it in MixEffect Settings > Automation > Enable HTTP Server.

**Comment by l4flirop** [comment]
Adam I seem to be running around in circles here 

<img width="2732" height="2048" alt="Image" src="https://github.com/user-attachments/assets/834dd740-dd14-472a-9df8-141033aa4fa7" />



<img width="1505" height="802" alt="Image" src="https://github.com/user-attachments/assets/45ba372c-a321-4e68-b65c-c872ea291ed5" />

**Comment by adamtow** [comment]
Your IP address in MixEffect Companion Module reads 10.100.20.21, whereas in MixEffect, the OSC server is 10.100.20.55. Use that in Companion.

**Comment by l4flirop** [comment]
SMH you know I was using the ip address for the switcher OMG. Adam thank you so very much Nice Big Fat Green Check Mark now 

<img width="1509" height="805" alt="Image" src="https://github.com/user-attachments/assets/f716c9d8-0f57-4b78-98cc-5b3bd35133e6" />
